### ISSUE TITLE
port to wayland

### ISSUE THREAD

**Issue by soloturn** [xmoto/xmoto#23]
this should be ported to wayland.

**Comment by Nikekson** [comment]
Seems to work in the [sdl2 branch](/xmoto/xmoto/tree/sdl2)

**Comment by soloturn** [comment]
cool, it works, with both ```SDL_VIDEODRIVER=x11``` and ```SDL_VIDEODRIVER=wayland```. the left-right-up-down keys for player1 brake-drive-flip left-flip right are displayed as some unicode chars though and do not work. setting them in options works. starting the sdl version displays the keys as "unknown".

**Comment by Nikekson** [comment]
Is this with a fresh xm.db file? The keysyms have completely changed from 1.2, and isn't only an issue on wayland

**Comment by soloturn** [comment]
true - no relation with X or wayland, but with sdl, sdl2. i only changed the dependencies in the arch PKGBUILD and compiled the sdl2 branch, and did not see a xm.db file, neither in .config nor in the generated package.

**Comment by Nikekson** [comment]
It's in ``~/.local/share/xmoto/xm.db``

**Comment by Nikekson** [comment]
The reason it happens is because the raw keysym values are stored in the xm.db file, it they're obviously invalid when moving to sdl2.
Also, as of now, this issue is the biggest obstacle holding back the migration

**Comment by soloturn** [comment]
looking at https://wiki.libsdl.org/MigrationGuide how SDL imagines that key preferences are stored then so migration would be straight forward?

**Comment by Nikekson** [comment]
The problem is - is there a way to do it without a version bump to xm.db?
Doing it would break compatibility, and overall the change feels too small to warrant that.

Another point to consider is that resetting the keys is not exactly optimal, but I can't think of any other way to do it without keeping the keysym defs around from an sdl1.2 header file, and writing mappings between the two.

The optimal middle ground would be to just reset the keybinds during startup, if it could somehow detect that they're from sdl1.2.
Another way would be to make the xm.db upgrades two-way, so you could downgrade too when needed, but this would likely become a huge maintenance burden, plus you would still need another mechanism for carrying it out (like a CLI argument)

**Comment by soloturn** [comment]
a version bump means to create a xm2.db file, which reads all settings from xm.db except the keys? or it means adding versioned properties inside the file with new names?

**Comment by Nikekson** [comment]
It's an sqlite db file, and has a table ``xm_parameters``, with a field ``xmdb_version``. A version bump means incrementing its value and running some upgrade routine in the ``xmDatabase::upgradeXmDbToVersion`` function ``db/xmDatabase.cpp``

**Comment by soloturn** [comment]
somehow i'd love, as a user, a backup db file. or 2 files, one for the old version, one for the new. so i can start the old version, just in case. without me noticing anything. as a programmer, if one could not detect that it is from 1.2 there is no way around bump the version, and make this explicit? having an upgrade and downgrade functionality is rolls royce - and would need some piece of documentation also - so - why bother?

**Comment by soloturn** [comment]
@Nikekson what is the state of your decision process concerning wayland?

**Comment by Nikekson** [comment]
It should be fully supported once the [sdl2 branch](/xmoto/xmoto/tree/sdl2) is merged, which I'm expecting to happen within a couple months, after some features are finished and all issues are verified to be gone (which are not specific to wayland)

**Comment by romulasry** [comment]
A new release would be great with sdl2.

**Comment by Nikekson** [comment]
Now that the SDL2 branch has been merged into master, the game should work under Wayland without major issues.
The only issue I'm currently aware of is broken resizing, but it is not implemented at all, so it's broken everywhere.

Leaving this open for reports.

**Comment by soloturn** [comment]
@Nikekson mind makeing a new release?
let me close this one ... everything lease should be a new bug report.
